src/robot/reporting/jsbuildingcontext.py

```diff
@@ -20,13 +20,16 @@
 from robot.utils import (attribute_escape, get_link_path, html_escape,
                          html_format, is_string, is_unicode, timestamp_to_secs,
                          unic)
+from robot.result.autoexpandkeywordmatcher import AutoExpandKeywordMatcher
```

Comment:
We try to avoid imports from other modules so that they need to go to submodules. Instead of `from robot.module.submodule import something` we try to do `from robot.module import something` i.e. expose `something` on the module root.

In this particular case it seems that the `autoexpandkeywordmatcher` module is in wrong place. It is under `robot.result` but seems to be used only under `robot.reporting` and that feels like a better "home" for it in general. Then the import could be like `from .autoexpandkeywordmatcher import xxx`.

Reply:
refactoring done